torch/jit/python_frontend.py

```diff
@@ -0,0 +1,285 @@
+import torch
+import sys
+import ast
+import inspect
+from functools import partial
+from collections import namedtuple
+from torch._C._jit_tree_views import *
+
+# TODO: improve error reporting (show source)
+#       once this is done, convert all asserts into checks with nicer error messages
+
+PY2 = sys.version_info[0] == 2
+_reserved_prefix = '__jit'
+
+
+def get_jit_ast(fn):
+    source = dedent(inspect.getsource(fn))
+    py_ast = ast.parse(source)
+    if len(py_ast.body) != 1 or not isinstance(py_ast.body[0], ast.FunctionDef):
+        raise RuntimeError("expected a single top-level function")
+    return build_def(SourceRangeFactory(source), py_ast.body[0])
+
+
+def dedent(source):
```

Comment:
`textwrap.dedent(text)`? https://docs.python.org/2/library/textwrap.html

Reply:
Nice, didn't know this one!